src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -176,6 +176,127 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_OFI_sigreq_complete(MPIR_Request ** sigreq)
     }
 }
 
+enum {
+    MPIDI_OFI_PUT,
+    MPIDI_OFI_GET,
+};
+
+static inline void MPIDI_OFI_load_iov(const void *buffer, int count, MPI_Datatype datatype,
+                                      MPI_Aint done, MPI_Aint max_len, MPI_Aint * offset,
+                                      struct iovec *iov)
+{
+    if (done == *offset) {
+        MPI_Aint outlen;
+        MPIR_Typerep_to_iov_offset(buffer, count, datatype, *offset, iov, max_len, &outlen);
+        *offset += outlen;
+    }
```

Comment:
This logic is a lot more cleaner and easier to follow that the previous draft.  Perhaps the variables can be a bit more descriptive?  How about:

`done` -> `current_iov_offset`
`offset` -> `loaded_iov_offset`

Reply:
Updated in latest version.